### PR TITLE
Expose fetchLimit and fetchOffset options

### PR DESCRIPTION
**Pull request by hishbase**
1. Expose `fetchLimit` and `fetchOffset` options on `NSFetchRequest` to `Database.fetch` function
2. Added unit tests

### CODE REVIEW

File: DatabasesTests/DatabasesTests.swift [hunk 1]
```diff
@@ -88,6 +88,52 @@ class DatabasesTests: XCTestCase {
...
     }
 
+    func testFetchLimit() throws {
+        let database = Database(type: .memory, modelURL: dbURL)
+
+        let currencies = [
+            TestCurrency(code: "JTC", name: "JOHNNYCOIN"),
+            TestCurrency(code: "ATC", name: "ANDREWCOIN"),
+            TestCurrency(code: "HTC", name: "HISHCOIN"),
+            ]
```
Comment (AndrewGold) [inline DatabasesTests/DatabasesTests.swift]: Nit: one less indent here

File: DatabasesTests/DatabasesTests.swift [hunk 2]
```diff
@@ -88,6 +88,52 @@ class DatabasesTests: XCTestCase {
...
+    }
+
+    func testFetchOffset() throws {
+        let database = Database(type: .memory, modelURL: dbURL)
+
+        let currencies = [
+            TestCurrency(code: "ATC", name: "ANDREWCOIN"),
+            TestCurrency(code: "HTC", name: "HISHCOIN"),
+            TestCurrency(code: "JTC", name: "JOHNNYCOIN"),
+            ]
```
Comment (AndrewGold) [inline DatabasesTests/DatabasesTests.swift]: Ditto

File: DatabasesTests/DatabasesTests.swift [hunk 3]
```diff
@@ -88,6 +88,52 @@ class DatabasesTests: XCTestCase {
...
+            .toBlocking(timeout: unitTestsTimeout)
+            .single()
+
+        let offsetResult: [TestCurrency] = try database.fetch(
+            sortDescriptors: sortDescriptors,
+            fetchOffset: 1,
+            fetchLimit: 1
+            )
+            .toBlocking(timeout: unitTestsTimeout)
+            .single()
```
Comment (AndrewGold) [inline DatabasesTests/DatabasesTests.swift]: Ditto on the above three lines

File: DatabasesTests/DatabasesTests.swift (hunk 1 above)
Comment (hishbase) [reply DatabasesTests/DatabasesTests.swift]: fixed

File: DatabasesTests/DatabasesTests.swift (hunk 2 above)
Comment (hishbase) [reply DatabasesTests/DatabasesTests.swift]: fixed

File: DatabasesTests/DatabasesTests.swift (hunk 3 above)
Comment (hishbase) [reply DatabasesTests/DatabasesTests.swift]: fixed

**Pull request merged**